mesos/framework/mirror_maker_task.go

```diff
@@ -0,0 +1,118 @@
+package framework
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/golang/protobuf/proto"
+	mesos "github.com/mesos/mesos-go/mesosproto"
+	util "github.com/mesos/mesos-go/mesosutil"
+)
+
+type MirrorMakerTask struct {
+	*CommonTask
+}
+
+func NewMirrorMakerTask(id string, queryParams url.Values) (*MirrorMakerTask, error) {
+	taskData := &TaskData{
+		ID:    id,
+		State: TaskStateInactive,
+		Config: TaskConfig{
+			"num.producers": "1",
+			"num.streams":   "1",
+			"queue.size":    "10000",
+		},
+	}
+
+	err := taskData.Update(queryParams)
+	if err != nil {
+		return nil, err
+	}
+
+	return &MirrorMakerTask{
+		&CommonTask{TaskData: taskData},
+	}, nil
+}
+
+func (mm *MirrorMakerTask) NewTaskInfo(offer *mesos.Offer) *mesos.TaskInfo {
+	taskName := fmt.Sprintf("mirrormaker-%s", mm.ID)
```

Comment:
I feel like we can abstract this out with another interface. MirrorMaker, "mirrormaker" has both consumer and producer and the log/metric configs. This staple broken up into the different common values and difference between the three make it so we can have a very plyable way to manage any consumer task running on Mesos. In fact, folks would just have to set their code up with theirs. I think in the struct interface we give them basically we are just having the config information, consumer, producer, consume+producer and the "strategy" functions (with error states).

After all of that it is all implementation side. The MirrorMaker code should be example for how to use Go Kafka Client to plug your code so that they run on mesos using the framework we built with cli, http, features, etc.